app/src/test/java/com/example/myapplication/PepeUnitTest.kt

```diff
@@ -0,0 +1,106 @@
+package com.example.myapplication
+
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+
+interface Bono {
+    fun bonoDesdeBasico(base: Double): Double
+}
+object BonoFijo: Bono {
+    override fun bonoDesdeBasico(base: Double): Double = 80.0
+}
+object BonoDiezPorciento: Bono {
+    override fun bonoDesdeBasico(base: Double): Double = base * 0.1
+}
+object BonoCero: Bono {
+    override fun bonoDesdeBasico(base: Double): Double = 0.0
+}
+
+interface Categoria {
+    fun basico(): Double
+}
+object Gerente: Categoria {
+    override fun basico(): Double = 30000.0
+}
+object Cadete: Categoria {
+    override fun basico(): Double = 15000.0
+}
+
+
+data class Empleado(var categoria: Categoria, var tipoDeBono: Bono) {
+    var faltas: Int = 0
+
+    fun basico(): Double = categoria.basico()
+
+    fun bono(): Double {
+        return tipoDeBono.bonoDesdeBasico(basico())
+    }
+
+    fun presentismo(): Int {
+        return 100 - faltas * 5
+    }
+
+    fun sueldo(): Double {
+        return basico() + bono() + presentismo()
+    }
+}
+
+class PepeUnitTest {
```

Comment:
Te desafío a que el único mensaje público de Pepe sea `sueldo`

Reply:
https://www.youtube.com/watch?v=4iOi_iPNC50

Reply:
[Best part ever](https://youtu.be/4iOi_iPNC50?t=38)